checks/raw/pinned_dependencies.go

```diff
@@ -498,3 +494,28 @@ var validateGitHubActionWorkflow fileparser.DoWhileTrueOnFileContent = func(
 
 	return true, nil
 }
+
+func isActionDependencyPinned(actionUses string) bool {
+	// Though GitHub Actions also allows actions to be specified as local filepaths
+	// (e.g. uses: <path/to/file/in/repo>), it's not yet practical to determine whether an
+	// arbitrary file that lives in the action context is a trustworthy, pinned workflow file
+	// So, those actions are treated as 'unpinned'
+
+	publicActionRegex := regexp.MustCompile(`^([\w-]+)/([\w-]+)(/([\w-/]+))?(@([a-f\d]+)|([\w\.-]+))?`)
+
+	if m := publicActionRegex.FindStringSubmatch(actionUses); len(m) > 0 {
+		ref := m[6]
```

Comment:
Does `6` correspond to the hash capture group? Maybe make that clear with a constant?

Is it possible for this to be out of bounds? You check there are capture groups, but not necessarily that there are more than 6.

I'm unfamiliar with the function, so it could be a guarantee provided by it
https://pkg.go.dev/regexp#Regexp.FindStringSubmatch

Reply:
From what I observed, if that group isn't captured, FindStringSubmatch will return an empty string at that submatch. The array size is fixed relative to groups defined in the pattern.